### PR TITLE
add my challenges as card in user profile

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: app/elements/coder/profile/user-challenges-table.html [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			</table>
+		</vaadin-grid>
+	</template>
+	<script type="application/javascript" src="/scripts/behaviors/providers/user-provider-behavior.js"></script>
+	<script>
+		// NOTE: members needs to be set for data to be shown.
+		Polymer({
+			is: 'user-challenges-table',
+			behaviors: [
+				Behaviors.Providers.User
```
Comment (pbochis) [inline app/elements/coder/profile/user-challenges-table.html]: why?

File: app/elements/coder/profile/profile-page.html [hunk 2]
```diff
@@ -119,7 +126,13 @@ <h3 class="card-header">Teams</h3>
 				</template>
 			</paper-material>
 			<paper-material class="right" elevation="1">
-				<h3 class="card-header">Challenges</h3>
+				<template is="dom-if" if="[[isCurrentUser]]">
```
Comment (pbochis) [inline app/elements/coder/profile/profile-page.html:129]: What if you remove the `dom-if` and just put `<user-challenges-table>` ? It's data provider will only recieve the visible challenges for the currently logged in user from the platform.

File: app/elements/coder/profile/user-challenges-table.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			observers: [
+				'_dataChanged(challenges)'
+			],
+			_selectedItemsChanged: function(){
+				var selected = this.$.challengesGrid.selection.selected();
+				if(!selected || selected.length === 0){
+					return;
+				}
+				var item = this.challenges[selected[0]];
+				if(item.status === 'COMPLETED'){
```
Comment (pbochis) [inline app/elements/coder/profile/user-challenges-table.html]: IMO I don't think it's nice UX to just redirect when a row is clicked in the table. Mby people randomly want to highlight something in a table.

File: app/elements/coder/profile/user-challenges-table.html [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+					page('/contest/' + item.challenge.id + '/result');
```
Comment (pbochis) [inline app/elements/coder/profile/user-challenges-table.html:55]: IMO I don't think it's nice UX to just redirect when a row is clicked in the table. Mby people randomly want to highlight something in a table.

Comment (victorbalan) [inline app/elements/coder/profile/user-challenges-table.html:55]: the mouse pointer indicates that the row is clickable.

File: app/elements/coder/profile/profile-page.html (hunk 2 above)
Comment (victorbalan) [inline app/elements/coder/profile/profile-page.html:129]: the platform stuff is not implemented for this and we would need another table for displaying that data so, the dom-if is ok.

**Pull request merged**